expression/aggregation/avg.go

```diff
@@ -102,24 +95,12 @@ func (af *avgFunction) calculateResult(ctx *aggEvaluateContext) (d types.Datum)
 	return
 }
 
-// GetGroupResult implements Aggregation interface.
-func (af *avgFunction) GetGroupResult(groupKey []byte) types.Datum {
-	ctx := af.getContext(groupKey)
+// GetResult implements Aggregation interface.
+func (af *avgFunction) GetResult(ctx *AggEvaluateContext) types.Datum {
```

Comment:
we can remove function `calculateResult()` and move the calculate logic to function `GetResult()`